Improve inactive group view

cascheberg opened the pull request:
### Contributor checklist

<!-- replace the empty checkboxes [ ] below with checked ones [x] accordingly -->
- [X] I am following the [Code Style Guidelines](https://github.com/WhisperSystems/Signal-Android/wiki/Code-Style-Guidelines)
- [X] I have tested my contribution
- [X] My contribution is fully baked and ready to be merged as is
- [X] I have made the choice whether I want the [BitHub reward](https://github.com/WhisperSystems/Signal-Android/wiki/BitHub-Rewards) or not by omitting or adding the word `FREEBIE` in my commit message

---
### Description

This PR improves **inactive group** and **blocked contact** conversation views:
* merge `initializeEnabledCheck()` and `setBlockedUserState()` into new `updateConversationState()`
* hide "disappearing messages" menu item
* hide call button
* hide "reset secure session"
* hide "invite" menu item
* hide compose panel in inactive groups
* rearrange menu items so that the first five and the last item are consistent for single and group conversations.


**Screenshot before** (single | active group | inactive group | blocked single)
![before](https://user-images.githubusercontent.com/3845150/59962587-c8e02600-94e7-11e9-81ae-142f659507f1.png)

**Screenshot after** (single | active group | inactive group | blocked single)
![after](https://user-images.githubusercontent.com/3845150/59962585-c54c9f00-94e7-11e9-80dc-54becaebb7e7.png)

// FREEBIE

scienmind commented:
Is there a reason to leave "Conversation Settings" and "Mute Notifications" for inactive groups?

(My understanding was that once you left the group, you won't receive anything there and hence muting/customizing sounds, etc. is not relevant there)

cascheberg commented:
@scienmind The one thing that I could think of was that someone repeatedly re-adds you to the group, so that you might want to mute it.

moxie0 commented on src/org/thoughtcrime/securesms/ConversationActivity.java:
```diff
@@ -1160,6 +1158,8 @@ private void setActionBarColor(MaterialColor color) {
   }
 
   private void setBlockedUserState(Recipients recipients) {
+    if (isGroupConversation()) return;
```
why was this necessary?

cascheberg replied:
`setBlockedUserState` is setting `composePanel.setVisibility`, which now conflicts with `initializeEnabledCheck`. This was sufficient to resolve the problem.

moxie0 replied:
Except now the method is no longer idempotent. Why don't you start with just the menu changes.

cascheberg replied:
I can do that. But I'd be glad if you proposed a different approach.

moxie0 commented on src/org/thoughtcrime/securesms/ConversationActivity.java:
```diff
@@ -1210,8 +1210,7 @@ private boolean isSelfConversation() {
   private boolean isGroupConversation() {
-    return getRecipients() != null &&
-        (!getRecipients().isSingleRecipient() || getRecipients().isGroupRecipient());
+    return getRecipients() != null && !isSingleConversation();
```
why is this necessary?

cascheberg replied:
This is to make clear how `isGroupConversation` behaves in relation to `isSingleConversation`, i.e. only one of them is true (which is different from how `isSingleRecipient` and `isGroupRecipient` work).

moxie0 commented:
looks nice

cascheberg commented:
Rebased and rewritten PR which checks for inactive groups and blocked contacts:
* merge `initializeEnabledCheck()` and `setBlockedUserState()` into new `updateConversationState()`
* hide "disappearing messages" menu item
* hide "reset secure session"
* hide "invite" menu item
* hide "add attachment" menu item

cascheberg commented:
Rebased and slightly improved with regard to #6400

cascheberg commented:
PR rebased, reduced amount of required changes, and tested again.

cascheberg commented:
PR rebased, tested and screenshots updated.

**Screenshot before** (single | active group | inactive group | blocked single)
![before](https://user-images.githubusercontent.com/3845150/59962587-c8e02600-94e7-11e9-81ae-142f659507f1.png)

**Screenshot after** (single | active group | inactive group | blocked single)
![after](https://user-images.githubusercontent.com/3845150/59962585-c54c9f00-94e7-11e9-80dc-54becaebb7e7.png)

cascheberg commented:
PR is outdated and issue has been fixed for the most part
